### PR TITLE
feat(eigen-client-extra-features): Add soft confirmations

### PR DESCRIPTION
**Pull request by gianbelinche**
## What ❔
This PR adds soft confirmations

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zkstack dev fmt` and `zkstack dev lint`.

### CODE REVIEW

File: core/node/da_clients/src/eigen/sdk.rs [hunk 1]
```diff
@@ -236,7 +237,16 @@ impl RawEigenClient {
                 disperser::BlobStatus::InsufficientSignatures => {
                     return Err(anyhow::anyhow!("Insufficient signatures"))
                 }
-                disperser::BlobStatus::Confirmed | disperser::BlobStatus::Finalized => {
+                disperser::BlobStatus::Confirmed => {
+                    println!("Blob dispatch confirmed");
```
Comment (juan518munoz) [inline core/node/da_clients/src/eigen/sdk.rs:241]: Remove `println!`, or replace it with a `tracing` log.

**Review by juan518munoz — APPROVED** [review]
LGTM. Be sure to check comment before merging!

File: core/node/da_clients/src/eigen/client.rs [hunk 2]
```diff
@@ -173,4 +173,34 @@ mod tests {
...
+                "d08aa7ae1bb5ddd46c3c2d8cdb5894ab9f54dec467233686ca42629e826ac4c6",
+            )
+            .unwrap(),
+        };
+        let client = EigenClient::new(config, secrets).await.unwrap();
+        let data = vec![1; 20];
+        let result = client.dispatch_blob(0, data.clone()).await.unwrap();
+        let blob_info: BlobInfo =
+            rlp::decode(&hex::decode(result.blob_id.clone()).unwrap()).unwrap();
+        // TODO: once get inclusion data is added, check it
```
Comment (juanbono) [inline core/node/da_clients/src/eigen/client.rs:202]: I think inclusion was already added.

Comment (gianbelinche) [reply core/node/da_clients/src/eigen/client.rs]: Done!

File: core/node/da_clients/src/eigen/client.rs [hunk 3]
```diff
@@ -203,4 +203,41 @@ mod tests {
...
+    #[tokio::test]
+    async fn test_wait_for_finalization() {
+        let config = EigenConfig::Disperser(DisperserConfig {
+            custom_quorum_numbers: None,
+            disperser_rpc: "https://disperser-holesky.eigenda.xyz:443".to_string(),
+            eth_confirmation_depth: -1,
+            eigenda_eth_rpc: String::default(),
+            eigenda_svc_manager_address: "0xD4A7E1Bd8015057293f0D0A557088c286942e84b".to_string(),
+            blob_size_limit: 2 * 1024 * 1024, // 2MB
+            status_query_timeout: 1800,       // 30 minutes
```
Comment (juanbono) [inline core/node/da_clients/src/eigen/client.rs:216]: is the configuration documented somewhere? I found confusing that the numbers in status_query_timeout and status_query_interval (both related to the status_query) have different units.

Comment (gianbelinche) [reply core/node/da_clients/src/eigen/client.rs]: Done

**Review by juanbono — APPROVED** [review]

File: core/node/da_clients/src/eigen/client.rs [hunk 4]
```diff
@@ -203,4 +203,41 @@ mod tests {
...
+            custom_quorum_numbers: None,
+            disperser_rpc: "https://disperser-holesky.eigenda.xyz:443".to_string(),
+            eth_confirmation_depth: -1,
+            eigenda_eth_rpc: String::default(),
+            eigenda_svc_manager_address: "0xD4A7E1Bd8015057293f0D0A557088c286942e84b".to_string(),
+            blob_size_limit: 2 * 1024 * 1024, // 2MB
+            status_query_timeout: 1800000,    // 30 minutes
+            status_query_interval: 5000,      // 5000 ms
+            wait_for_finalization: true,
+            authenticaded: true,
```
Comment (samlaf) [inline core/node/da_clients/src/eigen/client.rs:219]: typo: authenticaded -> authenticated

**Pull request merged**